sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,140 @@
+    }
+
+    /**
+     * Assigns an email address to the current internally tracked profile
      *
-     * @param email Email address for an active user
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param email Email address for active user
+     * @return
      */
-    fun setUserEmail(email: String) {
-        UserInfo.email = email
+    fun setEmail(email: String): Klaviyo = apply {
+        this.setProfile(Profile().setEmail(email))
     }
 
     /**
-     * Creates a track request for the Klaviyo APIs and begins processing.
-     * Track requests track the triggering of a specified [KlaviyoEvent] in Klaviyo
+     * Assigns a phone number to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param phoneNumber Phone number for active user
+     */
+    fun setPhoneNumber(phoneNumber: String): Klaviyo = apply {
+        this.setProfile(Profile().setPhoneNumber(phoneNumber))
+    }
+
+    /**
+     * Assigns an external ID to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
      *
-     * @param customerProperties A map of customer property information.
-     * Defines the customer that triggered this event
-     * @param properties A map of event property information.
-     * Additional properties associated to the event that are not for identifying the customer
+     * @param id Phone number for active user
+     * @return
      */
-    fun track(
-        event: KlaviyoEvent,
-        customerProperties: KlaviyoCustomerProperties,
-        properties: KlaviyoEventProperties? = null
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
+    fun setExternalId(id: String): Klaviyo = apply {
+        this.setProfile(Profile().setIdentifier(id))
     }
 
     /**
-     * Creates an identify request for the Klaviyo APIs and begins processing.
+     * Assign arbitrary attributes to the current profile by key
+     *
+     * This should be called when you collect additional data about your user
+     * (e.g. first and last name, or an address)
+     *
+     * @param propertyKey
+     * @param value
+     * @return
+     */
+    fun setProfileAttribute(propertyKey: KlaviyoProfileAttributeKey, value: String): Klaviyo =
+        setProfile(Profile().also { it[propertyKey] = value })
```

Comment:
should this also use the `apply` pattern

Reply:
doesn't need it, because setProfile already returns this. Actually I meant to take the `= apply` off all the other setters but `setProfile`.